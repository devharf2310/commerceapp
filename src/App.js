import React, { Component } from "react";
import { Route } from 'react-router-dom';
import NavigationDrawer from 'react-md/lib/NavigationDrawers';

import NavLink from './components/NavItemLink/NavItemLink';
import MenuTabs from './components/MenuTabs/MenuTabs';
import Routes from './components/Routes/Routes';
import Constants from './utils/Constants';
import ErrorBoundary from './pages/ErrorBoundary/ErrorBoundary';

import "./App.scss";

const { navItems } = Constants;


class App extends Component {
    handleVisibilityChange = () => { }
    render() {
     return (
        <ErrorBoundary>
            <Route
                render={({ location }) => (
                <NavigationDrawer
                    drawerTitle="Categories"
                    visible={location.pathname && location.pathname.startsWith("/products")}
                    onVisibilityChange={this.handleVisibilityChange}
                    toolbarChildren={<MenuTabs/>}
                    navItems={navItems.map(props => <NavLink {...props} key={props.to} />)}
                >
                <Routes location={location}/>
                </NavigationDrawer>
                )}
            />
        </ErrorBoundary>
    );
  }
}

 export default App;
