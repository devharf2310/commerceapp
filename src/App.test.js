import React from 'react';
import App from './App';
import renderer from 'react-test-renderer'
import MockRouter from 'react-mock-router';

window.matchMedia = window.matchMedia || function() {
    return {
        matches : false,
        addListener : function() {},
        removeListener: function() {}
    };
};


const location = { pathname: 'test' };
describe('app renders', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<MockRouter location={location}><App /></MockRouter>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
