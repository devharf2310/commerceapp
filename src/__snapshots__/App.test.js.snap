// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders renders correctly 1`] = `
<div>
  <header
    className="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--fixed"
  >
    <button
      className="md-btn md-btn--icon md-pointer--hover md-inline-block md-btn--toolbar md-toolbar--action-left"
      disabled={false}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-ink-container"
      />
      <i
        className="md-icon material-icons md-text--inherit"
      >
        menu
      </i>
    </button>
    <ul
      className="md-tabs documentation-tabs"
      role="tablist"
      style={
        Object {
          "paddingLeft": undefined,
        }
      }
    >
      <span
        aria-hidden={true}
      />
      <li
        aria-controls="documentation-panel-0"
        aria-pressed={false}
        aria-selected={true}
        className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--active"
        id="/"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="tab"
        tabIndex={0}
      >
        <div
          className="md-ink-container"
        />
        <div
          className="md-tab-label"
        >
          Home
        </div>
      </li>
      <li
        aria-controls="documentation-panel-1"
        aria-pressed={false}
        aria-selected={false}
        className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
        id="/products"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="tab"
        tabIndex={0}
      >
        <div
          className="md-ink-container"
        />
        <div
          className="md-tab-label"
        >
          Products
        </div>
      </li>
      <li
        aria-controls="documentation-panel-2"
        aria-pressed={false}
        aria-selected={false}
        className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
        id="/clients"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="tab"
        tabIndex={0}
      >
        <div
          className="md-ink-container"
        />
        <div
          className="md-tab-label"
        >
          Clients
        </div>
      </li>
      <li
        aria-controls="documentation-panel-3"
        aria-pressed={false}
        aria-selected={false}
        className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
        id="/contacts"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="tab"
        tabIndex={0}
      >
        <div
          className="md-ink-container"
        />
        <div
          className="md-tab-label"
        >
          Contact
        </div>
      </li>
      <span
        className="md-tab-indicator"
        style={
          Object {
            "MozTransform": "translate3d(0px, 0, 0)",
            "WebkitTransform": "translate3d(0px, 0, 0)",
            "height": 2,
            "msTransform": "translate3d(0px, 0, 0)",
            "transform": "translate3d(0px, 0, 0)",
            "width": 160,
          }
        }
      />
    </ul>
  </header>
  <main
    className="md-navigation-drawer-content md-navigation-drawer-content--inactive md-transition--acceleration md-toolbar-relative"
    id="main-content"
    tabIndex={-1}
  >
    <h1>
      Home
    </h1>
  </main>
</div>
`;
